projects/plugins/jetpack/modules/related-posts/jetpack-related-posts.php

```diff
@@ -174,14 +174,15 @@ public function get_headline() {
 	 * @returns string
 	 */
 	public function filter_add_target_to_dom( $content ) {
-		if ( has_block( 'jetpack/related-posts' ) ) {
+		if ( has_block( 'jetpack/related-posts' ) || Blocks::is_fse_theme() ) {
 			return $content;
 		}
 
 		if ( ! $this->_found_shortcode && ! doing_filter( 'get_the_excerpt' ) ) {
 			if ( class_exists( 'Jetpack_AMP_Support' ) && Jetpack_AMP_Support::is_amp_request() ) {
 				$content .= "\n" . $this->get_server_rendered_html();
 			} else {
+				$this->_enqueue_assets( true, true );
```

Comment:
It took me some time to understand the juggling that was happening when enqueuing assets. Do you think we could add some comments to make that clearer to our futureselves?

Reply:
Agreed! Thank you very much for the additional test / updates 💯